pkg/kds/server/server.go

```diff
@@ -0,0 +1,28 @@
+package server
+
+import (
+	"github.com/Kong/kuma/pkg/core"
+	core_runtime "github.com/Kong/kuma/pkg/core/runtime"
+	mads_server "github.com/Kong/kuma/pkg/mads/server"
```

Comment:
It's definitely not a good idea for `kds` to depend on `mads`.
A little copy pasting from `mads/server/components` is fine.

Reply:
I thought maybe make sense to move that functionality to `pkg/util/xds`?

Reply:
To be honest, after reading `mads/server/components` I don't think it's worth it, but I'll leave it to you for your judgement :)

Reply:
Okay since we don't have much time, copy-pasting is good for now :)